xcdat/spatial.py

```diff
@@ -309,25 +310,26 @@ def _validate_axis_arg(self, axis: List[SpatialAxis]):
             # Check the axis coordinate variable exists in the Dataset.
             get_axis_coord(self._dataset, key)
 
-    def _validate_domain_bounds(self, domain_bounds: xr.DataArray):
-        """Validates the ``domain_bounds`` arg based on a set of criteria.
-
+    def _force_domain_order_low_to_high(self, domain_bounds: xr.DataArray):
+        """Re-orders the ``domain_bounds`` to be ordered low-to-high such that
+        domain_bounds[:, 0] < domain_bounds[:, 1].
         Parameters
         ----------
         domain_bounds: xr.DataArray
-            The bounds of an axis
-
-        Raises
+            The bounds of an axis.
+        Returns
         ------
-        TypeError
-            If the ``domain_bounds`` of a grid cell are not ordered low-to-high.
+        xr.DataArray
+            The bounds of an axis (re-ordered if applicable).
         """
         index_bad_cells = np.where(domain_bounds[:, 1] - domain_bounds[:, 0] < 0)[0]
         if len(index_bad_cells) > 0:
-            raise ValueError(
-                "The bounds have unexpected ordering. A lower bound has a "
-                "greater value than an upper bound."
-            )
+            new_domain_bounds = domain_bounds.copy()
+            new_domain_bounds[index_bad_cells, 0] = domain_bounds[index_bad_cells, 1]
+            new_domain_bounds[index_bad_cells, 1] = domain_bounds[index_bad_cells, 0]
+            return new_domain_bounds
+        else:
+            return domain_bounds
```

Comment:
Previously, we threw an error if the lower domain bound was larger than the upper domain bound (in `_validate_domain_bounds`) This function is replaced with `_force_domain_order_low_to_high` which re-orders the bounds such that `domain_bound[:, 0] < domain_bounds[:, 1]`.

Note I had originally replaced [these lines](https://github.com/xCDAT/xcdat/blob/main/xcdat/spatial.py#L601-L602) with logic to simply determine, which of the bounds is higher versus lower (e.g., `domain_lowers = np.min(d_bounds, axis=1)`), but the order matters further down ([here](https://github.com/xCDAT/xcdat/blob/main/xcdat/spatial.py#L619)).

Reply:
Looks good